src/hooks/UrqlUseMutation.re

```diff
@@ -22,12 +22,17 @@ external useMutationJs:
   (~query: string) => (useMutationResponseJs('a), executeMutation) =
   "useMutation";
 
-let useMutationResponseToRecord = (result: useMutationResponseJs('a)) => {
-  let data = result->dataGet;
+let useMutationResponseToRecord =
+    (parse: Js.Json.t => 'response, result: useMutationResponseJs('a)) => {
+  let data =
+    switch (result->dataGet) {
+    | None => None
+    | Some(d) => Some(parse(d))
+    };
   let error = result->errorGet;
   let fetching = result->fetchingGet;
 
-  let response: UrqlTypes.response('a) =
+  let response: UrqlTypes.response('response) =
     switch (fetching, data, error) {
     | (true, _, _) => Fetching
```

Comment:
One more "similar to `useQuery`" item – I liked how you stripped specifying the type here and instead supplied `UrqlTypes.Fetching` on the first match of the pattern matching (L37) to locally open that variant type. Let's opt for that here as well 🙌